### ISSUE TITLE
Random errors with a lot of pathfinding

### ISSUE THREAD

**Issue by SuperMax8** [olijeffers0n/PatheticAPI#82]
### Stack trace

```
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:11] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:11] [ForkJoinPool-7-worker-3/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:11] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:11] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-11/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-7/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-6/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-3/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-7/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-14/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-2/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:11] [ForkJoinPool-7-worker-3/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:11] [ForkJoinPool-7-worker-3/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] A fatal error has occurred: Failed to find path async
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] Please open an issue on the Pathetic GitHub page with all this information:
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] Version: 0.1.0
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] Server Version: git-Purpur-2176 (MC: 1.20.4)
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] Java Version: 21.0.3
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:12] [ForkJoinPool-7-worker-5/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-15/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] OS: Linux
[20:33:12] [ForkJoinPool-7-worker-1/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] OS Architecture: amd64
[20:33:12] [ForkJoinPool-7-worker-12/ERROR]: [org.patheloper.util.ErrorLogger] ===============================
```

### Actions to reproduce (if known)

doing lot of pathfinding in the same area, I'm making a full packet mob system with custom ai, and when I've a lot of mobs following me I got these errors

the cpu usage also sky rocket randomly
![image](https://github.com/patheloper/pathetic/assets/73855070/0ac6730a-87b6-420a-9d11-c2831a7445cd)

**Comment by bsommerfeld** [comment]
Hey, thanks for reporting this issue!
In order to help us better understand and resolve the problem, could you please share a code snippet where this error occurs? Any additional details or context you can provide would be greatly appreciated.

**Comment by SuperMax8** [comment]
this is my test code
```java
public void tickPathFinding() {
        Location currentLoc = getLocation();
        target = WorldUtils.getClosestPlayer(currentLoc, 30, p -> p.getGameMode().equals(GameMode.SURVIVAL));
        if (target == null) {
            path.clear();
            return;
        }
        Location targetLoc = target.getLocation();
        if (lastTargetLocation != null && currentLoc.distanceSquared(targetLoc) <= 1 && targetLoc.getX() == lastTargetLocation.getX() && targetLoc.getY() == lastTargetLocation.getY() && targetLoc.getZ() == lastTargetLocation.getZ())
            return;
        PathFinderUtils.findPhysicEntityPath(currentLoc, target.getLocation(), this, points -> {
            path = points;
            lastTargetLocation = targetLoc.clone();
            if (!path.isEmpty()) path.removeFirst();
            /*target.sendMessage("Targeting");*/
        });
    }


public static void findPhysicEntityPath(Location from, Location to, AbstractPhysicsEntity entity, Consumer<List<Location>> whenCompleted) {
        try {
            CompletionStage<PathfinderResult> result = pathfind(from, to, new WalkablePathfinderStrategy());
            result.thenAccept(pathResult -> {
                if (!pathResult.successful()) {
                    whenCompleted.accept(new ArrayList<>());
                    return;
                }
                LinkedList<Location> list = new LinkedList<>();
                for (PathPosition pos : pathResult.getPath()) list.add(BukkitMapper.toLocation(pos).add(0.5, 0, 0.5));
                whenCompleted.accept(simplifyPath(list, 3));
            });
        } catch (Exception e) {
            e.printStackTrace();
        }
    }


public static CompletionStage<PathfinderResult> pathfindCompletion(Location from, Location to, PathfinderStrategy strategy) {
        return pathfinder.findPath(BukkitMapper.toPathPosition(from), BukkitMapper.toPathPosition(to), strategy);
    }
```



I tick a lot the function in async when I test with a lot of entities following me, and it randomly send errors and stop working
I'm on purpur 1.20.4 with latest pathetic

**Comment by bsommerfeld** [comment]
Thanks for sharing, i will investigate ASAP and keep you up to date in here.

**Comment by bsommerfeld** [comment]
This has been fixed and will be contained in the next update of pathetic

**Comment by bsommerfeld** [comment]
Fixed in [2.4.6](https://github.com/patheloper/pathetic/releases/tag/2.4.6)
